frontend/src/metabase/redux/requests.js

```diff
@@ -80,9 +80,12 @@ function requestStateReducerRecursive(state, action) {
 
 export default (state = {}, action) => {
   if (action && action.payload && action.payload.statePath) {
-    state = updateIn(state, action.payload.statePath, subState =>
-      requestStateReducerRecursive(subState, action),
-    );
+    const statePath = action.payload.statePath;
+    if (statePath.length > 2 || getIn(state, statePath)) {
+      state = updateIn(state, action.payload.statePath, subState =>
+        requestStateReducerRecursive(subState, action),
+      );
+    }
```

Comment:
Could you please explain what exactly the check does here? Wondering what the `2` number means here. I think it'd make sense to maybe extract a condition to a constant with a more human-friendly name or maybe adding a comment or something

Reply:
For sure! There are 2 types of invalidations: specific request invalidations (which include a key/id in the state path), and bulk invalidations which just say "invalidate all requests for this entity". Those bulk invalidations have a statePath that is 2 items long. The fix as it exists right now is to ensure that there is something there to invalidate, which is why we also call `getIn(state, statePath)`

Reply:
Awesome! Yeah, I'll make this more readable and include a comment in there for future reference